main_test.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/vouch/vouch-proxy/pkg/cfg"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func Test_listen_uds(t *testing.T) {
+	// socket file does not yet exist
+	{
+		tempDir, err := os.MkdirTemp("", "")
+		assert.NoError(t, err)
+		socketFile := filepath.Join(tempDir, "socket0")
+		config := &cfg.Config{
+			Listen: fmt.Sprintf("unix:%s", socketFile),
+		}
+		lis, cleanupFn, err := listen(config)
```

Comment:
instead of contructing a config object, please create and consume a config file in `config/testing` and then use a setup function similar to...
https://github.com/vouch/vouch-proxy/blob/master/pkg/cfg/cfg_test.go#L23-L26

please also add config examples in `config/config.yml_example`  and `config/config.yml_example_listen_unix_socket`

Reply:
- Updated test to use testing config files instead of constructing a config object
- Added config examples to `config/config.yml_example`. I didn't add another standalone one, because it would just be an exact duplicate of another one of the other files with one extra line.